### PR TITLE
Fixes a slew of bounds issues.

### LINKED ISSUE #104: Panick when cat-ing a torrent file
With the following torrent http://releases.ubuntu.com/16.04/ubuntu-16.04.1-desktop-amd64.iso.torrent when I do
```
cat ubuntu-16.04.1-desktop-amd64.iso.torrent
```
I get
```
[unhandled] execute byte=89
[unhandled] execute byte=04
[unhandled] execute byte=03
[unhandled] execute byte=12
[unimplemented] move_down_and_cr: 1
[unhandled] execute byte=11
[unhandled] execute byte=8c
thread 'pty reader' panicked at 'index out of bounds: the len is 24 but the index is 28', /Users/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-mac/build/src/libcollections/vec.rs:1371
stack backtrace:
   1:        0x10eafdfaa - std::sys::imp::backtrace::tracing::imp::write::h1461f67be4bf68a9
   2:        0x10eb0018f - std::panicking::default_hook::{{closure}}::h6dba65b2b498d274
   3:        0x10eaffe37 - std::panicking::default_hook::ha8ecbcff1f59e805
   4:        0x10eb00656 - std::panicking::rust_panic_with_hook::h5ceed7bd7aef311a
   5:        0x10eb004a4 - std::panicking::begin_panic::h091a980eb2ddd29a
   6:        0x10eb00412 - std::panicking::begin_panic_fmt::h063a49cc71e3d103
   7:        0x10eb00377 - rust_begin_unwind
   8:        0x10eb28530 - core::panicking::panic_fmt::h59374b8a88ef3536
   9:        0x10eb284a8 - core::panicking::panic_bounds_check::h78cc4243bb390438
  10:        0x10ea5b7da - <alacritty::ansi::Performer<'a, H, W> as vte::Perform>::print::he49de47c781c0e03
  11:        0x10ea53808 - <alacritty::event_loop::EventLoop<Io>>::spawn::{{closure}}::hac156881432fe821
  12:        0x10ea4162b - std::panicking::try::do_call::h2c88261f55e7a440
  13:        0x10eb0152a - __rust_maybe_catch_panic
  14:        0x10ea516ba - <F as alloc::boxed::FnBox<A>>::call_box::he707c721e5ef5370
  15:        0x10eaff864 - std::sys::imp::thread::Thread::new::thread_start::haeb202749da670ee
  16:     0x7fff8a8dd99c - _pthread_body
  17:     0x7fff8a8dd919 - _pthread_start
child finished with error '256'
```

It maybe because it's not valid UTF8?

At 8d3f5f4e179b6fe55617dc72fe60fa1bdfd9b014

Thank you!

### LINKED ISSUE #108: Thai text displayed incorrectly
I work with multilingual data. I was excited to have a terminal that would not slow to a pathetic crawl just because it was asked to show some words in Thai, and saw that Alacritty was doing pretty well at keeping up the frame rate in uncooperative languages.

Two problems, though, which might be related:

* The text rendering in Thai doesn't look right. I think the characters are taking up more character cells than they're supposed to, wrapping in unexpected places, and generally getting smeared all over the screen.
* After a few seconds of displaying Thai text, it crashes.

I'm running on Ubuntu 16.04, and my font is Unifont.

Here's the backtrace:

```
thread 'pty reader' panicked at 'index out of bounds: the len is 231 but the index is 234', /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcollections/vec.rs:1371
stack backtrace:
   1:     0x55e76d0b8fca - std::sys::imp::backtrace::tracing::imp::write::h2b860f17b6b0f511
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:42
   2:     0x55e76d0be33f - std::panicking::default_hook::{{closure}}::hedea7d569eb1e6b8
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:349
   3:     0x55e76d0bdf3e - std::panicking::default_hook::he87e0c76c89c52a4
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:365
   4:     0x55e76d0be7e7 - std::panicking::rust_panic_with_hook::h304127bde2d3d5e2
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:553
   5:     0x55e76d0be674 - std::panicking::begin_panic::h814f60faa3ce8314
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:515
   6:     0x55e76d0be599 - std::panicking::begin_panic_fmt::hf4f7d700fa077693
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:499
   7:     0x55e76d0be527 - rust_begin_unwind
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:475
   8:     0x55e76d0ea5ed - core::panicking::panic_fmt::hee207ff21fb62166
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/panicking.rs:69
   9:     0x55e76d0ea598 - core::panicking::panic_bounds_check::hca28eae069525a56
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/panicking.rs:56
  10:     0x55e76cfb4572 - <alacritty::ansi::Performer<'a, H, W> as vte::Perform>::print::hc3f37ea7fe292a15
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcollections/vec.rs:1371
                        at /home/rspeer/src/alacritty/src/grid.rs:311
                        at /home/rspeer/src/alacritty/src/grid.rs:225
                        at /home/rspeer/src/alacritty/src/term/mod.rs:638
                        at /home/rspeer/src/alacritty/src/ansi.rs:438
  11:     0x55e76cfad00d - <alacritty::event_loop::EventLoop<Io>>::spawn::{{closure}}::h48d857cbcad6c3a4
                        at /home/rspeer/.cargo/registry/src/github.com-1ecc6299db9ec823/vte-0.1.2/src/lib.rs:167
                        at /home/rspeer/.cargo/registry/src/github.com-1ecc6299db9ec823/vte-0.1.2/src/lib.rs:137
                        at /home/rspeer/src/alacritty/src/ansi.rs:85
                        at /home/rspeer/src/alacritty/src/event_loop.rs:211
                        at /home/rspeer/src/alacritty/src/event_loop.rs:308
  12:     0x55e76cf9868b - std::panicking::try::do_call::h0f1252efbc6ce7c2
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panic.rs:295
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:458
  13:     0x55e76d0c572a - __rust_maybe_catch_panic
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libpanic_unwind/lib.rs:98
  14:     0x55e76cfaac3a - <F as alloc::boxed::FnBox<A>>::call_box::h356923938038b1e5
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:434
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panic.rs:351
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/thread/mod.rs:287
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/liballoc/boxed.rs:605
  15:     0x55e76d0bd634 - std::sys::imp::thread::Thread::new::thread_start::h460a0adc61139b06
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/liballoc/boxed.rs:615
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/sys_common/thread.rs:21
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/sys/unix/thread.rs:84
  16:     0x7f50f3fc36b9 - start_thread
  17:     0x7f50f3ae382c - clone
  18:                0x0 - <unknown>
child finished with error '1'
```

The Thai text I've got is just sampled from the Common Crawl, so I can point you to the exact file I'm trying to display:

https://s3.amazonaws.com/uploads.hipchat.com/11494/33245/TSThk9XemC0LX6B/thai-example.txt

### LINKED ISSUE #116: `ls` at ubuntu 16.10 crashes
`RUST_BACKTRACE=1 ./alacritty` outputs that once running `ls`:

```
thread 'pty reader' panicked at 'index out of bounds: the len is 2 but the index is 2', /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/slice.rs:640
stack backtrace:
   1:     0x55f57d115dea - std::sys::imp::backtrace::tracing::imp::write::h2b860f17b6b0f511
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:42
   2:     0x55f57d11b15f - std::panicking::default_hook::{{closure}}::hedea7d569eb1e6b8
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:349
   3:     0x55f57d11ad5e - std::panicking::default_hook::he87e0c76c89c52a4
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:365
   4:     0x55f57d11b607 - std::panicking::rust_panic_with_hook::h304127bde2d3d5e2
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:553
   5:     0x55f57d11b494 - std::panicking::begin_panic::h814f60faa3ce8314
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:515
   6:     0x55f57d11b3b9 - std::panicking::begin_panic_fmt::hf4f7d700fa077693
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:499
   7:     0x55f57d11b347 - rust_begin_unwind
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:475
   8:     0x55f57d14740d - core::panicking::panic_fmt::hee207ff21fb62166
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/panicking.rs:69
   9:     0x55f57d1473b8 - core::panicking::panic_bounds_check::hca28eae069525a56
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/panicking.rs:56
  10:     0x55f57d02df5d - alacritty::ansi::parse_color::he1f954b5b4f73a83
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/slice.rs:640
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/slice.rs:542
                        at /home/chrmod/Projects/opensource/alacritty/src/ansi.rs:732
  11:     0x55f57cff689c - vte::Parser::perform_action::ha595f6d3713fa1a4
                        at /home/chrmod/Projects/opensource/alacritty/src/ansi.rs:630
                        at /home/chrmod/.cargo/registry/src/github.com-1ecc6299db9ec823/vte-0.1.2/src/lib.rs:205
  12:     0x55f57d009aba - <alacritty::event_loop::EventLoop<Io>>::spawn::{{closure}}::h8a87f59e0d85f278
                        at /home/chrmod/.cargo/registry/src/github.com-1ecc6299db9ec823/vte-0.1.2/src/lib.rs:158
                        at /home/chrmod/.cargo/registry/src/github.com-1ecc6299db9ec823/vte-0.1.2/src/lib.rs:137
                        at /home/chrmod/Projects/opensource/alacritty/src/ansi.rs:85
                        at /home/chrmod/Projects/opensource/alacritty/src/event_loop.rs:211
                        at /home/chrmod/Projects/opensource/alacritty/src/event_loop.rs:308
  13:     0x55f57cff518b - std::panicking::try::do_call::h36ad92b485fdc90c
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panic.rs:295
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:458
  14:     0x55f57d12254a - __rust_maybe_catch_panic
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libpanic_unwind/lib.rs:98
  15:     0x55f57d00797a - <F as alloc::boxed::FnBox<A>>::call_box::h9e0ff91ec3235773
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:434
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panic.rs:351
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/thread/mod.rs:287
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/liballoc/boxed.rs:605
  16:     0x55f57d11a454 - std::sys::imp::thread::Thread::new::thread_start::h460a0adc61139b06
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/liballoc/boxed.rs:615
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/sys_common/thread.rs:21
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/sys/unix/thread.rs:84
  17:     0x7f896b4fe709 - start_thread
  18:     0x7f896b0210ae - __clone
  19:                0x0 - <unknown>
child finished with error '1'
```

### LINKED ISSUE #133: Crash on printf '\e[81C '
When running `printf '\e[81C\eH'`:

```
thread 'pty reader' panicked at 'index out of bounds: the len is 80 but the index is 81', /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcollections/vec.rs:1371
stack backtrace:
   1:     0x5588ad085d4c - std::sys::imp::backtrace::tracing::imp::write::h9c41d2f69e5caabf
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:42
   2:     0x5588ad08b23e - std::panicking::default_hook::{{closure}}::h1f61f3c769fffe7a
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:351
   3:     0x5588ad08ae44 - std::panicking::default_hook::hd5bda4e453dfb4be
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:367
   4:     0x5588ad08b6e7 - std::panicking::rust_panic_with_hook::hffbc74969c7b5d87
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:555
   5:     0x5588ad08b574 - std::panicking::begin_panic::h3bbdbb2a23440035
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:517
   6:     0x5588ad08b499 - std::panicking::begin_panic_fmt::h34f5b320b0f94559
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:501
   7:     0x5588ad08b427 - rust_begin_unwind
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:477
   8:     0x5588ad0b75bd - core::panicking::panic_fmt::h1016b85b51d1931f
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/panicking.rs:69
   9:     0x5588ad0b7568 - core::panicking::panic_bounds_check::h3b875b6ea9a674f6
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/panicking.rs:56
  10:     0x5588accec545 - <collections::vec::Vec<T> as core::ops::IndexMut<usize>>::index_mut::h69b598477d3e900a
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcollections/vec.rs:1371
  11:     0x5588acc5fa44 - <alacritty::grid::Row<T> as core::ops::IndexMut<alacritty::index::Column>>::index_mut::h9adf91b5b10bafe3
                        at /home/anders/rust/alacritty/src/grid.rs:311
  12:     0x5588acc64280 - <alacritty::grid::Grid<T> as core::ops::IndexMut<&'point alacritty::index::Point>>::index_mut::h2a8363d21414e0ce
                        at /home/anders/rust/alacritty/src/grid.rs:225
  13:     0x5588accdaef7 - <alacritty::term::Term as alacritty::ansi::Handler>::input::h1bacbff20d66275a
                        at /home/anders/rust/alacritty/src/term/mod.rs:638
  14:     0x5588accf6d6c - <alacritty::ansi::Performer<'a, H, W> as vte::Perform>::print::h2f59924c294298c6
                        at /home/anders/rust/alacritty/src/ansi.rs:438
  15:     0x5588acc80616 - vte::Parser::perform_action::h82162452c9ac296a
                        at /home/anders/.cargo/registry/src/github.com-1ecc6299db9ec823/vte-0.1.2/src/lib.rs:189
  16:     0x5588acc80c4e - vte::Parser::perform_state_change::ha318e6e99fbda378
                        at /home/anders/.cargo/registry/src/github.com-1ecc6299db9ec823/vte-0.1.2/src/lib.rs:167
  17:     0x5588acc80fc4 - vte::Parser::advance::h66c6bd0670baf377
                        at /home/anders/.cargo/registry/src/github.com-1ecc6299db9ec823/vte-0.1.2/src/lib.rs:137
  18:     0x5588accfbfac - alacritty::ansi::Processor::advance::h92e3d5ba20d53d25
                        at /home/anders/rust/alacritty/src/ansi.rs:85
  19:     0x5588accc6986 - <alacritty::event_loop::EventLoop<Io>>::pty_read::h18a7cc99c622d42f
                        at /home/anders/rust/alacritty/src/event_loop.rs:211
  20:     0x5588accc6082 - <alacritty::event_loop::EventLoop<Io>>::spawn::{{closure}}::h069a2120b2d00625
                        at /home/anders/rust/alacritty/src/event_loop.rs:308
  21:     0x5588accf8024 - <std::panic::AssertUnwindSafe<F> as core::ops::FnOnce<()>>::call_once::h6032dcb0cf344702
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panic.rs:296
  22:     0x5588acc8048f - std::panicking::try::do_call::hc745aec8a9cedc1b
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:460
  23:     0x5588ad09265a - __rust_maybe_catch_panic
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libpanic_unwind/lib.rs:98
  24:     0x5588acc7fb03 - std::panicking::try::h697849a0f4444602
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:436
  25:     0x5588acc7dbb4 - std::panic::catch_unwind::hb5b74bacab7a9eb7
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panic.rs:361
  26:     0x5588acc7f45c - std::thread::Builder::spawn::{{closure}}::hc2b962b0b41adef6
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/thread/mod.rs:357
  27:     0x5588accbf2e9 - <F as alloc::boxed::FnBox<A>>::call_box::hab7d1897638a2589
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/liballoc/boxed.rs:605
  28:     0x5588ad08a534 - std::sys::imp::thread::Thread::new::thread_start::h76badbf9b0ecaf58
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/liballoc/boxed.rs:615
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/sys_common/thread.rs:21
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/sys/unix/thread.rs:84
  29:     0x7f76efc8c6c9 - start_thread
  30:     0x7f76ef7af0ae - __clone
  31:                0x0 - <unknown>
child finished with error '1'
```

### PR DESCRIPTION
**Pull request by mbrumlow**
Fixed a bunch of bounds issues. Most of them were caused by input in src/term/mod.rs leaving self.cursor.col dirty after printing the last char on a line before a wrap was forced. 

These changes ensure when using self.cursor.col and self.cursor.line are within sane values before using them when using them when handling other escape sequences. Care is also taken to ensure the resulting updates to self.cursor.col and self.cursorline are within proper bounds. 

#63 - Fixed
#104 - Fixed 
#108 - Fixed 
#116 - Fixed
#133 - Fixed

### CODE REVIEW

**Comment by jwilm** [comment]
Thanks for putting this together! I'm going to merge this for now since it fixes a bunch of issues, but I'd like to circle back because it appears some performance is being left on the table (basically some `limit` calls could be coalesced).

File: src/term/mod.rs [hunk 1]
```diff
@@ -580,15 +581,16 @@ impl Term {
...
+            let end = limit(start + lines, Line(0), self.grid.num_lines() - 1);
             self.grid.clear_region(start..end, |c| c.reset(&template));
         }
 
         // Scroll from origin to bottom less number of lines
         {
-            let start = origin;
-            let end = self.scroll_region.end - lines;
+            let start = limit(origin, Line(0), self.grid.num_lines() - 1);
+            let end = limit(self.scroll_region.end - lines, Line(0), self.grid.num_lines() - 1);
```
Comment (jwilm) [inline src/term/mod.rs]: I think that some of these can be avoided by ensuring things like the `scroll_region` are valid when being set and after resizing.

**Comment by mbrumlow** [comment]
I agree 100%. If some things are moved around we can avoid many of the limits. There are two key places that are of concern. 

1) Anything coming from the pty has to be sanitized to the terminal sizes. That is why cat /dev/urandom caused so many problems. 

2) If the schematics to know if the next time input is added needs to be wrapped were changed and did not rely self.cursor.col == elf.grid.num_cols().  Fixing this would mean we can avoid cleaning self.cursor.col when combined from arguments for escape sequences that need that value for its calculation.

**Comment by jwilm** [comment]
I agree with your diagnosis and was already thinking about #2--fixing that would allow us to remove many limits.

Do you fancy continuing to work on this? I'd like to make some progress through the issue tracker, but that precludes actually writing code.

File: src/term/mod.rs [hunk 2]
```diff
@@ -691,28 +697,29 @@ impl ansi::Handler for Term {
...
-        self.cursor.line -= lines;
+        self.cursor.line = limit(self.cursor.line - lines, Line(0), self.grid.num_lines() -1);
     }
 
     #[inline]
     fn move_down(&mut self, lines: Line) {
         debug_println!("move_down: {}", lines);
-        self.cursor.line += lines;
+        self.cursor.line = limit(self.cursor.line + lines, Line(0), self.grid.num_lines() - 1);
+        debug_println!("move_down - > lines: {}", self.cursor.line);
```
Comment (mbrumlow) [inline src/term/mod.rs]: I ment to remove that, sorry, will have a update shortly with that gone.

**Comment by vks** [comment]
@jwilm Could you close #63, #108 and #116? I believe they were fixed by this PR.

@mbrumlow If you write `Fixes #63`, then Github will automatically close issue #63 once the commit is merged.

**Comment by jwilm** [comment]
#63 is not quite resolved, and I'm not yet confident about #108. Updating #116 (although it was not solved in this patch)

**Comment by vks** [comment]
I cannot reproduce #63 (maybe it did not run for long enough to crash) or #108.

**Comment by jwilm** [comment]
Catting `/dev/urandom` now takes longer to exhibit any issues, but there's still two distinct panics that show up given enough time.

Hoping to hear back from the author of #108 that the issue is resolved.

**Comment by mbrumlow** [comment]
I am trying to reproduce the panic on cat /dev/urandom, I let it run for about 60 seconds before, i will let it run longer.

**Comment by jwilm** [comment]
@mbrumlow the two issues I still encounter are have backtraces [in this comment](https://github.com/jwilm/alacritty/issues/63#issuecomment-271103638). Running latest master on my mbp and still get a panic.

**Pull request merged**